frontend/front/src/pages/Admin/user/CreateNewUser.js

```diff
@@ -1,149 +1,209 @@
-import { Box, Button, TextField, Typography } from "@mui/material";
-import React from "react";
+import {
+  Box,
+  Button,
+  Container,
+  Modal,
+  Stack,
+  Typography,
+} from "@mui/material";
+import {
+  ROLE_ADMIN,
+  ROLE_SURVEYOR,
+  generatePassword,
+} from "../../../features/login/loginUtils";
+import React, { useCallback, useState } from "react";
+import {
+  useCreateSurveyorMutation,
+  useCreateUserMutation,
+} from "../../../api/apiSlice";
+
+import CustomSnackbar from "../../../components/CustomSnackbar";
+import { HeatPumpDropdown } from "../../../components/SurveyComponent/HeatPumpDropdown";
+import { HeatPumpPhoneField } from "../../../components/SurveyComponent/HeatPumpPhoneField";
+import { HeatPumpTextField } from "../../../components/SurveyComponent/HeatPumpTextField";
+import Loader from "../../../components/Loader";
+import { useForm } from "react-hook-form";
 import { useNavigate } from "react-router-dom";
-import { useForm, Controller } from "react-hook-form";
+
+const ACTION_BACK = "BACK";
+const ACTION_NEW = "NEW";
 
 const CreateNewUser = () => {
   const navigate = useNavigate();
-  const { handleSubmit, reset, control } = useForm({
+
+  const [modalData, setModalData] = useState(null);
+  const [userError, setUserError] = useState(null);
+  const [isUserLoading, setIsUserLoading] = useState(false);
+
+  const { handleSubmit, control, reset } = useForm({
     defaultValues: {
-      firstName: "",
-      lastName: "",
+      firstname: "",
+      lastname: "",
       email: "",
       phone: "",
-      streetAddress: "",
+      street_address: "",
       city: "",
-      zipCode: "",
       state: "",
+      zipcode: "",
+      role: ROLE_SURVEYOR,
     },
   });
-  const onSubmit = (data) => console.log(data);
 
-  const handleCancel = () => {
+  const [createUser] = useCreateUserMutation();
+  const [createSurveyor] = useCreateSurveyorMutation();
+
+  const handleCancel = useCallback(() => {
     navigate("/admin/user");
-  };
-  return (
-    <Box
-      display="flex"
-      justifyContent="center"
-      alignItems="center"
-      flexDirection="column"
-    >
-      <Box width={500} mt={5}>
-        <Box sx={{ bgcolor: "primary.main", color: "white" }} p={1}>
-          <Typography variant="h5">Create User Profile</Typography>
+  }, [navigate]);
+
+  const onSubmit = useCallback(
+    async (data) => {
+      setIsUserLoading(true);
+
+      const password = generatePassword();
+
+      try {
+        const user = await createUser({
```

Comment:
The lifecycle here is:
1. create devise user
2. user id to create surveyor
3. show the current user a modal with the email and auto-generated password to be passed along to the surveyor

Not 100% on how we should handle this, another option would be to give each user the same password and then let people reset the passwords individually (something I think we should do anyway).

Reply:
Decision was to have Admin write passwords down and then have users change their passwords until we get email set up on the backend.